src/cloudai/workloads/ai_dynamo/ai_dynamo.sh

```diff
@@ -79,6 +85,74 @@ _parse_cli_pairs() {
   done
 }
 
+_set_backend_defaults() {
+  case "${dynamo_args["backend"]}" in
+    vllm)
+      :
+      ;;
+    sglang)
+      dynamo_args["prefill-cmd"]="python3 -m dynamo.sglang.worker"
+      dynamo_args["decode-cmd"]="python3 -m dynamo.sglang.decode_worker"
+      dynamo_args["ingress-cmd"]="python3 -m dynamo.frontend"
+      ;;
+    *)
+      log "ERROR: Unknown backend '${dynamo_args["backend"]}'"
+      exit 1
+      ;;
+  esac
+}
+
+_is_vllm() { [[ "${dynamo_args["backend"]}" == "vllm" ]]; }
+_is_sglang() { [[ "${dynamo_args["backend"]}" == "sglang" ]]; }
+
+_sync_num_nodes_from_section_args() {
+  if [[ -n "${prefill_args["--num-nodes"]:-}" ]]; then
+    dynamo_args["num-prefill-nodes"]="${prefill_args["--num-nodes"]}"
+  fi
+  if [[ -n "${decode_args["--num-nodes"]:-}" ]]; then
+    dynamo_args["num-decode-nodes"]="${decode_args["--num-nodes"]}"
+  fi
```

Comment:
In one of the changes to my run.sh, I removed worker.num_nodes in favor of dynamo.num_prefill_nodes and dynamo.num_decode_nodes. This reduces complexity of handling `--prefill` and `--decode` flags.